pennylane/drawer/circuit_drawer.py

```diff
@@ -62,6 +63,9 @@ def __init__(
         max_length=None,
         _label_offsets=None,
     ):
+        warnings.warn(
+            "CircuitDrawer is deprecated. Please see qml.drawer.tape_text instead.", UserWarning
```

Comment:
@albi3ro quick question: what do you think about overloading `qml.draw()` so that it accepts tapes and qnodes?

E.g.,

```python
>>> qml.draw(tape)
>>> qml.draw(qnode)(*args, **kwargs)
```

Reply:
This means less overhead for developers/users needing to find the correct function, at the expense of a more complicated docstring.

Reply:
looking into it, see #2281